modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -1105,6 +1105,26 @@ public void visit(ParameterDef paramDef) {
 
     @Override
     public void visit(VariableDef varDef) {
+        // Resolves the type of the variable
```

Comment:
This code block should not be in this method. This logic is not generic to all variable definitions